drivers/eks/eks_driver.go

```diff
@@ -989,6 +1007,14 @@ func (d *Driver) Update(ctx context.Context, info *types.ClusterInfo, options *t
 		state.ClientSecret = newState.ClientSecret
 	}
 
+	if newState.DesiredASGSize > 0 && newState.DesiredASGSize != state.DesiredASGSize {
```

Comment:
Can you move the newState.DesiredASGSize to the validate() function and have validate work for updates and create or create a separate function?